AsyncDisplayKit/ASVideoNode.h

```diff
@@ -0,0 +1,30 @@
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+typedef NS_ENUM(NSUInteger, ASVideoGravity) {
+  ASVideoGravityResizeAspect,
+  ASVideoGravityResizeAspectFill,
+  ASVideoGravityResize
+};
+
+@interface ASVideoNode : ASDisplayNode<_ASDisplayLayerDelegate>
+@property (atomic, strong, readwrite) AVAsset *asset;
+@property (nonatomic, assign, readwrite) BOOL shouldAutoPlay;
+@property (atomic) ASVideoGravity gravity;
+@property (atomic) BOOL autorepeat;
+@property (atomic) ASButtonNode *playButton;
+
+- (void)play;
+- (void)pause;
+
+@end
+
+@protocol ASVideoNodeDelegate <NSObject>
+@end
+
+@protocol ASVideoNodeDatasource <NSObject>
+@optional
+- (ASDisplayNode *)playButtonForVideoNode:(ASVideoNode *) videoNode;
+- (UIImage *)thumbnailForVideoNode:(ASVideoNode *) videoNode;
+- (NSURL *)thumbnailURLForVideoNode:(ASVideoNode *)videoNode;
```

Comment:
So I put these in here at first but then just added the property for the playButton and figured we could figure out the direction we wanted to go later.  Do you think we should just switch all these to properties?  Is that kind of the standard in the framework so far?